packages/spanner/src/database.js

```diff
@@ -233,6 +244,12 @@ Database.formatName_ = function(instanceName, name) {
  */
 Database.prototype.close = function(callback) {
   var self = this;
+  var leaks = this.pool_.getSessionLeaks();
+
+  if (leaks.length) {
+    callback(new SessionLeakError(leaks));
+    return;
```

Comment:
You should still clear the pool and delete all the sessions before returning here.

Reply:
How should we proceed here? Merge, then improve, or Improve, then merge?

Reply:
I don't really have the bandwidth at the moment, so I would vote to merge then improve. I think I'd like to swap out pooling libraries at some point which could also play a role in fixing this.

Reply:
So if I understand correctly, if there is a leak (even a single session), then none of the sessions will get deleted? That's quite bad especially since it seem to me that node.js is more prone to session leaks due to its async nature. I would rather we keep track of all sessions ids somewhere and delete them all ourselves if pooling library will not do this.